internal/pkg/gateway/apiutils.go

```diff
@@ -57,9 +59,44 @@ func newRpcError(code codes.Code, message string, details ...proto.Message) erro
 	return st.Err()
 }
 
-func wrappedRpcError(err error, message string, details ...proto.Message) error {
-	statusErr := status.Convert(err)
-	return newRpcError(statusErr.Code(), message+": "+statusErr.Message(), details...)
+func createTx(
+	header *common.Header,
+	proposalPayload []byte,
+	resps ...*peer.ProposalResponse,
+) (*common.Envelope, error) {
+	// ensure that all actions are bitwise equal
+	payload := resps[0].Payload
+	for _, r := range resps {
+		if !bytes.Equal(payload, r.Payload) {
+			return nil, errors.New("ProposalResponsePayloads do not match")
+		}
+	}
+
+	var endorsements []*peer.Endorsement
```

Comment:
Can we please not revert the logic that was put in place in this [PR](https://github.com/hyperledger/fabric/pull/3040)? I am referring to the de-duplication.

Let's de-duplicate it like in that PR if you wish to move the logic here.

Reply:
Sure - I hadn't fully appreciated why that code was there until I read your linked PR.